### ISSUE TITLE
Images downloaded does not have the GPS Data

### ISSUE THREAD

**Issue by hongning123** [andyjsmith/SmugMug-Downloader#17]
I found downloaded images does not have the GPS Data that the original pic have.  Looking thru your code I notice you use the LargestImage, is there a reason why you didn't use ImageDownload.   ImageDownload seems to download the original pic with the GPS data.  I am thinking I am missing something.

**Comment by andyjsmith** [comment]
Where do you see "ImageDownload"? I don't see that anywhere in the API spec, but I also haven't worked with it for a long time. Can you provide a publicly-accessible example?

**Comment by hongning123** [comment]
i check it thru api browser

https://api.smugmug.com/api/v2/album/dCR8Zd/image/RdXNChX-0

look thru the Json data and you will see ImageDownload its below the
largestimage

**Comment by andyjsmith** [comment]
I don't see that, see below. I'm guessing based on your username that this is your own album. So I bet that's only available if you're logged in and you're the owner of the album. (In that case -- you should be able to download the album yourself on the Smugmug website, you shouldn't need to use this tool)
```json
"Uris": {
    "LargestImage": {
        "Uri": "/api/v2/image/RdXNChX-0!largestimage",
        "Locator": "LargestImage",
        "LocatorType": "Object",
        "UriDescription": "Largest size available for image",
        "EndpointType": "LargestImage"
    },
    "ImageSizes": {
        "Uri": "/api/v2/image/RdXNChX-0!sizes",
        "Locator": "ImageSizes",
        "LocatorType": "Object",
        "UriDescription": "Sizes available for image",
        "EndpointType": "ImageSizes"
    },
    "ImageSizeDetails": {
        "Uri": "/api/v2/image/RdXNChX-0!sizedetails",
        "Locator": "ImageSizeDetails",
        "LocatorType": "Object",
        "UriDescription": "Detailed size information for image",
        "EndpointType": "ImageSizeDetails"
    },
    "PointOfInterest": {
        "Uri": "/api/v2/image/RdXNChX-0!pointofinterest",
        "Locator": "PointOfInterest",
        "LocatorType": "Object",
        "UriDescription": "Point of interest for image",
        "EndpointType": "PointOfInterest"
    },
    "PointOfInterestCrops": {
        "Uri": "/api/v2/image/RdXNChX-0!poicrops",
        "Locator": "PointOfInterestCrops",
        "LocatorType": "List",
        "UriDescription": "PointOfInterest Crops for image",
        "EndpointType": "PointOfInterestCrops"
    },
    "Regions": {
        "Uri": "/api/v2/image/RdXNChX-0!regions",
        "Locator": "Region",
        "LocatorType": "Objects",
        "UriDescription": "Regions for image",
        "EndpointType": "Regions"
    },
    "ImageComments": {
        "Uri": "/api/v2/image/RdXNChX-0!comments",
        "Locator": "Comment",
        "LocatorType": "Objects",
        "UriDescription": "Comments on image",
        "EndpointType": "ImageComments"
    },
    "ImageMetadata": {
        "Uri": "/api/v2/image/RdXNChX-0!metadata",
        "Locator": "ImageMetadata",
        "LocatorType": "Object",
        "UriDescription": "Metadata for image",
        "EndpointType": "ImageMetadata"
    },
    "ImagePrices": {
        "Uri": "/api/v2/image/RdXNChX-0!prices",
        "Locator": "CatalogSkuPrice",
        "LocatorType": "Objects",
        "UriDescription": "Purchasable Skus",
        "EndpointType": "ImagePrices"
    },
    "ImagePricelistExclusions": {
        "Uri": "/api/v2/image/RdXNChX-0!pricelistexclusions",
        "Locator": "ImagePricelistExclusions",
        "LocatorType": "Object",
        "UriDescription": "Pricelist information for an image",
        "EndpointType": "ImagePricelistExclusions"
    },
    "Album": {
        "Uri": "/api/v2/album/dCR8Zd",
        "Locator": "Album",
        "LocatorType": "Object",
        "UriDescription": "Album by key",
        "EndpointType": "Album"
    },
    "Image": {
        "Uri": "/api/v2/image/RdXNChX-0",
        "Locator": "Image",
        "LocatorType": "Object",
        "UriDescription": "Image by key",
        "EndpointType": "Image"
    },
    "AlbumImagePricelistExclusions": {
        "Uri": "/api/v2/album/dCR8Zd/image/RdXNChX-0!pricelistexclusions",
        "Locator": "AlbumImagePricelistExclusions",
        "LocatorType": "Object",
        "UriDescription": "Pricelist information for an album image",
        "EndpointType": "AlbumImagePricelistExclusions"
    },
    "AlbumImageMetadata": {
        "Uri": "/api/v2/album/dCR8Zd/image/RdXNChX-0!metadata",
        "Locator": "AlbumImageMetadata",
        "LocatorType": "Object",
        "UriDescription": "Metadata for AlbumImage",
        "EndpointType": "AlbumImageMetadata"
    },
    "AlbumImageShareUris": {
        "Uri": "/api/v2/album/dCR8Zd/image/RdXNChX-0!shareuris",
        "Locator": "AlbumImageShareUris",
        "LocatorType": "Object",
        "UriDescription": "URIs that are useful for sharing",
        "EndpointType": "AlbumImageShareUris"
    }
},
```

**Comment by hongning123** [comment]
true its my own album,  i use it as ,the download of smugmug is tedious.  i
had to use sessionid to make it work.   so you are right if your not the
owner i believe likelihood you wont have access to original pictures. i
just preferred the original as it has the gps attribute.  Thanks for
explaining.

**Comment by andyjsmith** [comment]
Can you try the dev branch here and let me know if that fixed it for you? I don't have a SM account so I can't test. https://github.com/andyjsmith/SmugMug-Downloader/tree/dev

**Comment by hongning123** [comment]
I will try that.

**Comment by hongning123** [comment]
Yep it's working if you have rights to it (session id) it will do
ImageDownload else's LargestImage.   Thanks

**Comment by andyjsmith** [comment]
Fixed #19
